batch/sql/migration-1.sql

```diff
@@ -0,0 +1,461 @@
+ALTER TABLE `attempts` ADD CONSTRAINT fk_instance_name
+  FOREIGN KEY (instance_name) REFERENCES instances(name) ON DELETE CASCADE;
+
+ALTER TABLE instances ADD time_activated BIGINT;
+ALTER TABLE instances ADD time_deactivated BIGINT;
+ALTER TABLE instances ADD removed BOOLEAN NOT NULL DEFAULT FALSE;
+CREATE INDEX `instances_removed` ON `instances` (`removed`);
```

Comment:
I realized I may not have fully resolved the migration problem, at least for batch.

First, I think we should take down the service we're migrating so and old version isn't modifying the database while we're doing the migration (or after).  I'll work on this.  There will be a short outage, but unless we're modifying lots of data, the migration should be very fast.  Long term, I think we need to do the upgrade in stages and only allow migrations that are fast and intact nicely with the old and new version (e.g. adding or removing an unused column).

However, batch is harder, because if the migration takes the batch driver down, it isn't there to run the deploy step to bring itself back up.  For now, I think re-deploying batch by hand after the migration is finished is probably the easiest thing.

Finally, we need to worry about how pre-existing data will be handled.  Is batch OK with instances with NULL activation time?  I'm guessing the right thing to do is delete the pre-existing instances.  Note, I think that will mean most attempts going forward will have an instance, but not all until we're purging older jobs (e.g. older than 3 months).